components/sensor/ultrasonic/ultrasonic.go

```diff
@@ -99,32 +99,34 @@ func newSensor(ctx context.Context, deps registry.Dependencies, name string, con
 		s.timeoutMs = 1000
 	}
 
-	s.intChan = make(chan board.Tick)
-	s.readingChan = make(chan bool)
-	s.errChan = make(chan error)
-	s.distanceChan = make(chan float64)
+	s.ticksChan = make(chan board.Tick, 2)
 	if err := s.triggerPin.Set(ctx, false, nil); err != nil {
 		return nil, errors.Wrap(err, "ultrasonic: cannot set trigger pin to low")
 	}
-	s.startUpdateLoop(ctx)
+	s.echoInterrupt.AddCallback(s.ticksChan)
```

Comment:
I think you're better off moving this to the `Readings` call. My concern is that with timeouts (and other flakiness) that we've seen on these sensors, if there's a rogue interrupt on this channel it can contaminate future readings. Having a new channel + callback is created per `Readings` call may make it more robust. 
I don't feel too strongly about this...wdyt?

Reply:
I like Steve's idea. Noise on the wire will be ignored, and it will be at least as correct as the current version. and the `BasicDigitalInterrupt`'s implementation of `AddCallback` and `RemoveCallback` are threadsafe, so you can call `Readings` from multiple threads simultaneously without trouble.

Put in a comment explaining the idea, though!